jax/interpreters/pxla.py

```diff
@@ -266,23 +307,53 @@ def _map(f, *xs):
   return tuple(map(f, *xs))
 
 
+class ShardedDeviceTuple(xla.DeviceTuple):
```

Comment:
I think this deserves a comment describing what it is because I am not going to remember the contents of the PR description in a months' time.

Reply:
You're absolutely right. In fact, I don't remember what's going on and I just wrote this yesterday...

I just pushed some docstrings, PTAL! We might want to land-and-iterate if these are good enough as a first cut.